tests/model_monitoring/test_applications/test_app_context.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+
+import pytest
+
+from mlrun.model_monitoring.applications.context import MonitoringApplicationContext
+from mlrun.projects import MlrunProject
+
+
+@pytest.mark.parametrize("method", ["log_artifact", "log_dataset", "log_model"])
+def test_log_object_signature(method: str) -> None:
+    """Future-proof the `log_x` method of MM app context with respect to the project object"""
+    assert inspect.signature(
+        getattr(MonitoringApplicationContext, method)
+    ) == inspect.signature(getattr(MlrunProject, method))
```

Comment:
Nice.
Maybe add a test that actually makes sure that when logging artifacts through the context, they're created with the expected labels, and belong to the project as expected?

Reply:
This is covered in:
* `tests/model_monitoring/test_application_steps.py::TestEventPreparation::test_prepare_monitoring_event` - a new unit test, with the labels check, not exactly with the project. I'll add the project check to it.
* The existing system tests you commented on - with the project but without checking the labels. I will add the labels check to them.